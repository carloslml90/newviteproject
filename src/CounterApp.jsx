import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp=({value}) => {
console. log ('executing')
    const [counter, SetCounter]=useState(value);

    const CounterAdd=() => { 

        SetCounter(

        (c) => c + 1);
    }

    const CounterSubstract=() => {

        SetCounter(

            (c) => counter -1
        );
    }    

    const CounterReset=() => {

        SetCounter(

            (c) => value
        );
    }    




    return (
    <>
        <h1>Counter APP</h1>

       

        <h2> Cups of coffe today {counter}</h2>
        <button onClick={CounterAdd}> +1 </button>
        <button onClick={CounterSubstract}> -1 </button>
        <button onClick={CounterReset}> Reset </button>

    </> 
   
        )
}

CounterApp.propTypes={

value: PropTypes.number.isRequired,
}

