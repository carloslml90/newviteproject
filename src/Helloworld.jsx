

import PropTypes from 'prop-types';
export const Hello =({title, description, goal}) => {
 const  NewFunction=() =>{
console.log('hola mundo');
}
return (
<>
    <h1 data-testid="test-title">{title}</h1>
    <p>{goal}</p>
    <p>{description}</p>


    </>
)



}


Hello.propTypes= {
title: PropTypes.string,
description: PropTypes.string,
goal: PropTypes.string





}