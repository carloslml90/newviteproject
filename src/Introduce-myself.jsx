


const name= 'carlos';

const apellido='mata'

const fullname= name.toUpperCase()+' '+ apellido.toUpperCase();

const dates={age:35, stature: 5.7 }


export const MyIntroduce=() =>{


    return ( 

<p> Hello my name is {fullname} and my data are the following: <code>{JSON.stringify(dates)}</code> </p>

        )
    



}