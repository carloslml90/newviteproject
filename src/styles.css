p {font-size: 5 em ; font-family: Verdana, Geneva, Tahoma, sans-serif; font-style: oblique; color: rosybrown };