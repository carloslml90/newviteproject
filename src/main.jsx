import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App'
import { Hello } from './Helloworld'
import { MyIntroduce } from './Introduce-myself'
import { CounterApp } from './CounterApp'


import "./styles.css"
/* import './index.css' */



ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={50} />

 <Hello 
 title='i love me a lot'
 description='nothing interesting'
 goal='Will be a great progammer'/>
 
  </React.StrictMode>,
)
