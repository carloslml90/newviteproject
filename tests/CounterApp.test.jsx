import { render, screen} from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Testing my <CounterApp/>', ()=>{

const InitialValue= 20
test('It must does match with a snapshot',()=>{

const{container}=render(<CounterApp value={InitialValue}/>);

expect(container).toMatchSnapshot();

})



 test('it must show a start value of 100',()=>{

render(<CounterApp value={100}/>);
expect(screen.getByText('100')).toBeTruthy();

}) 

})

