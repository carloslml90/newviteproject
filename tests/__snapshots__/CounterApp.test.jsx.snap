// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing my <CounterApp/> It must does match with a snapshot 1`] = `
<div>
  <h1>
    Counter APP
  </h1>
  <h2>
     Cups of coffe today 
    20
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
